src/utils/path-cache.ts

```diff
@@ -78,7 +78,7 @@ export class PathCache {
 
     let pathExists: boolean;
     while (
-      !(pathExists = existsSync(join(basePath, aliasPathPart))) &&
+      !(pathExists = this.exists(join(basePath, aliasPathPart))) &&
```

Comment:
existsResolvedAlias uses the cache, but does that also resolve the bug?
```suggestion
      !(pathExists = this.existsResolvedAlias(join(basePath, aliasPathPart))) &&
```

Reply:
It does also resolve the bug.

However,  this is inside getAAP, which is only called in getAbsoluteAliasPath, which has the cache already, so I think it would be redundant.

Let me know if you still want me to merge the suggestion.

Reply:
Another thought, would you rather I add a `resolveFiles` switch, defaulting to false, to make this non-breaking?

Reply:
This solution is probably better.
Using this.exists is fine.